### PR TITLE
[SPARK-29381][FOLLOWUP][PYTHON][ML] Add 'private' _XXXParams classes for classification & regression

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add private _XXXParams classes for classification & regression

### Why are the changes needed?
To keep parity between scala and python

### Does this PR introduce any user-facing change?
Yes. Add gettters/setters for the following Model classes

```
LinearSVCModel:
get/setRegParam
get/setMaxIte
get/setFitIntercept
get/setTol
get/setStandardization
get/setWeightCol
get/setAggregationDepth
get/setThreshold

LogisticRegressionModel:
get/setRegParam
get/setElasticNetParam
get/setMaxIter
get/setFitIntercept
get/setTol
get/setStandardization
get/setWeightCol
get/setAggregationDepth
get/setThreshold

NaiveBayesModel:
get/setWeightCol

LinearRegressionModel:
get/setRegParam
get/setElasticNetParam
get/setMaxIter
get/setTol
get/setFitIntercept
get/setStandardization
get/setWeight
get/setSolver
get/setAggregationDepth
get/setLoss

GeneralizedLinearRegressionModel:
get/setFitIntercept
get/setMaxIter
get/setTol
get/setRegParam
get/setWeightCol
get/setSolver
```

### How was this patch tested?
Add a few doctest

### CODE REVIEW

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -271,10 +282,167 @@ def intercept(self):
...
+
+    @since("1.5.0")
+    def setThresholds(self, value):
+        """
+        Sets the value of :py:attr:`thresholds`.
+        Clears value of :py:attr:`threshold` if it has been set.
+        """
+        self._set(thresholds=value)
+        self._clear(self.threshold)
+        return self
```
Comment (huaxingao) [inline python/pyspark/ml/classification.py]: it's a little strange to have ```setThreshold/Thresholds``` in the XXXParams class, but scala ```LogisticRegressionParams``` does this way, so I just do the same to be consistent with scala side.

**Review by srowen — COMMENTED** [review]
Are there any actual changes to the API or it's just moving definitions around?

**Comment by zhengruifeng** [comment]
@srowen This PR should add above-mentioned methods. (Some of them are redundant setters, we will remove in following PR)

@huaxingao some conflicts happened

**Comment by huaxingao** [comment]
@srowen sorry, I somehow didn't see your comment. As @zhengruifeng mentioned, this PR adds setters/getters in ```LinearSVCModel```, ```LogisticRegressionModel```, ```NaiveBayesModel```, 
```LinearRegressionModel``` and ```GeneralizedLinearRegressionModel```. The redundant setters will be remove later in the last parity PR.

@zhengruifeng I will rebase and resolve the conflict.

**Review by zhengruifeng — APPROVED** [review]

**Comment by zhengruifeng** [comment]
merged to master, thanks all

**Comment by huaxingao** [comment]
Thanks! @zhengruifeng @srowen

**Pull request closed without merging**